### ISSUE TITLE
Implement CORN loss for logit outputs

### ISSUE THREAD

**Issue by gmgeorg** [ck37/coral-ordinal#15]
The coral-pytorch authors recently published an update on CORAL ordinal regression, called CORN ordinal regression.  tl;dr: it's a stick breaking process implementation of ordinal regression, which means no weight constraints are necessary to keep rank consistency.

See here for template
https://github.com/Raschka-research-group/coral-pytorch/blob/main/coral_pytorch/losses.py#L86

TF keras implementation can be completely analogous using `tf.math.log_sigmoid()`.
